core/src/main/java/org/apache/iceberg/deletes/Deletes.java

```diff
@@ -227,6 +237,39 @@ public void close() {
     }
   }
 
+  private static class PositionStreamDeleteMarker<T> extends PositionStreamDeleteFilter<T> {
```

Comment:
as mentioned in the draft PR, this is not actually a filter as it always returns true. It is only to use the iteration part of  the filter. Maybe use `CloseableIterable#transform` instead for iteration?

Reply:
Yeah, I've replace `filter` with `transform` for all places except this one. This is trickier to change. Let me think a bit more.

Reply:
Refactored